### PR TITLE
fix: oas2 collectionFormat when not 'multi'

### PR DESCRIPTION
**Pull request by tim-lai**
### Description

Fix OAS2 collectionFormat handling for non-multipart cases
```
collectionFormat: csv
collectionFormat: tsv
collectionFormat: ssv
collectionFormat: pipes
```

### Motivation and Context

Previous OAS2 multipart fixes did not cover non-multi cases.

### Types of changes

- [ ] No code changes (changes to documentation, CI, metadata, etc)
- [ ] Dependency changes (any modification to dependencies in `package.json`)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: test/http-multipart.js [hunk 1]
```diff
@@ -42,6 +47,22 @@ describe('buildRequest - openapi 2.0', () => {
       expect(itemEntries.length).toEqual(2)
       expect(itemEntries[0]).toEqual('person1')
       expect(itemEntries[1]).toEqual('person2')
+      // set of non-multi collectionFormat cases below
```
Comment (char0n) [inline test/http-multipart.js:50]: I'd suggest to return to my original comment https://github.com/swagger-api/swagger-js/pull/1527#discussion_r423757494 about how we approach building tests here. We're just adding stuff into the test and making it bigger and more complex. IMHO building tests in this manner will not scale for future and maintainability.

File: src/http.js [hunk 2]
```diff
@@ -214,7 +221,7 @@ function buildFormData(reqForm) {
    * @return {FormData} - new FormData instance
    */
   return Object.entries(reqForm).reduce((formData, [name, input]) => {
-    if (!input.collectionFormat && !input.isOAS3formatArray) {
+    if (!(input.collectionFormat && input.collectionFormat === 'multi') && !input.isOAS3formatArray) {
```
Comment (char0n) [inline src/http.js:224]: More readable suggestion
```suggestion
    if (has(input, 'collectionFormat') && input.collectionFormat !== 'multi' && !input.isOAS3formatArray) {
```
[has](https://lodash.com/docs/4.17.15#has) function comens from `lodash`

File: test/http-multipart.js (hunk 1 above)
Comment (tim-lai) [reply test/http-multipart.js:50]: done. separated and categorized 'expects"

File: src/http.js (hunk 2 above)
Comment (char0n) [reply src/http.js:224]: I think my suggestion is logically wrong because I misunderstood the negated conjunction.

Comment (char0n) [inline src/http.js:224]:
```suggestion
    if ((isNil(input.collectionFormat) || input.collectionFormat !== 'multi') && !input.isOAS3formatArray) {
```

**Pull request merged**